day_7/10_speckled_frogs.rb

```diff
@@ -0,0 +1,86 @@
+### Required
+num_array = [10,9,8,7,6,5,4,3,2,1]
+num_array.each do |number|
+  frog_nouns = ""
+  last_line = ""
+  end_punctuation = ""
+  frogs_left = ""
+  if number == 1
+    frog_nouns = "frog"
+    last_line = "were no more"
+    frogs_left = ""
+    end_punctuation = "!"
+  elsif number == 2
+    frog_nouns = "frogs"
+    last_line = "was"
+    frogs_left = "#{number - 1} "
+    end_punctuation = "."
+  else
+    frog_nouns = "frogs"
+    last_line = "were"
+    frogs_left = "#{number - 1} "
+    end_punctuation = "."
+  end
+  puts "#{number} speckled #{frog_nouns} sat on a log
+  eating some most delicious bugs.
+  One jumped in the pool where its nice and cool,
+  then there #{last_line} #{frogs_left}speckled frogs#{end_punctuation}"
```

Comment:
Great recognition of where to pull out shared words in this string and use your iteration to only change the interpolated values.